[Issue]: CPU Backend - Link Failure

### Problem Description

```
make
[ 98%] Built target modules
[100%] Linking CXX shared library lib/librpp.dylib
Undefined symbols for architecture x86_64:
  "_ffts_execute", referenced from:
      spectrogram_host_tensor(float*, RpptDesc*, float*, RpptDesc*, int*, bool, bool, float*, int, int, int, int, rpp::Handle&) (.omp_outlined) in spectrogram.cpp.o
  "_ffts_free", referenced from:
      spectrogram_host_tensor(float*, RpptDesc*, float*, RpptDesc*, int*, bool, bool, float*, int, int, int, int, rpp::Handle&) (.omp_outlined) in spectrogram.cpp.o
  "_ffts_init_1d", referenced from:
      spectrogram_host_tensor(float*, RpptDesc*, float*, RpptDesc*, int*, bool, bool, float*, int, int, int, int, rpp::Handle&) (.omp_outlined) in spectrogram.cpp.o
  "_ffts_init_1d_real", referenced from:
      spectrogram_host_tensor(float*, RpptDesc*, float*, RpptDesc*, int*, bool, bool, float*, int, int, int, int, rpp::Handle&) (.omp_outlined) in spectrogram.cpp.o
ld: symbol(s) not found for architecture x86_64
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/librpp.2.0.0.dylib] Error 1
make[1]: *** [CMakeFiles/rpp.dir/all] Error 2
make: *** [all] Error 2

```

### Operating System

macOS

### CPU

ANY

### GPU

AMD Instinct MI300

### Other

_No response_

### ROCm Version

ROCm 6.0.0

### ROCm Component

rpp

### Steps to Reproduce

_No response_

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

_No response_

### Additional Information

_No response_

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -335,7 +335,7 @@ set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${PROJECT_VERSION} SOVE
 target_include_directories(${PROJECT_NAME}
     PUBLIC
         ${CMAKE_CURRENT_SOURCE_DIR}/api
-        ${CMAKE_CURRENT_SOURCE_DIR}/third_party/ffts/include
+        $<$<BOOL:${RPP_AUDIO_SUPPORT}>:${CMAKE_CURRENT_SOURCE_DIR}/third_party/ffts/include>
         ${ROCM_PATH}/include
     PRIVATE
         ${CMAKE_CURRENT_SOURCE_DIR}/src/include/cpu

--- a/src/modules/CMakeLists.txt
+++ b/src/modules/CMakeLists.txt
@@ -99,7 +99,7 @@ function(remove_audio_augmentations SOURCE_LIST BACKEND)
         "${CMAKE_CURRENT_SOURCE_DIR}/tensor/cpu/kernel/resample.cpp"
         "${CMAKE_CURRENT_SOURCE_DIR}/tensor/cpu/kernel/mel_filter_bank.cpp"
     )
-    set(${SOURCE_LIST} "${${SOURCE_LIST}}" PARENT_SCOPE)
+    set(${SOURCE_LIST} ${${SOURCE_LIST}} PARENT_SCOPE)
 endfunction()
 
 function(add_fma_property)

--- a/src/modules/CMakeLists.txt
+++ b/src/modules/CMakeLists.txt
@@ -185,6 +185,7 @@ elseif( "${BACKEND}" STREQUAL "OCL")
     file(GLOB MOD_CL_CPP "batch_pd/cl/*.cpp" )
     file(GLOB MOD_HOST_CPP_TENSOR_KERNELS "tensor/cpu/kernel/*.cpp" )
     list(APPEND Rpp_Source ${CPPFILES} ${MOD_HOST_CPP_TENSOR_KERNELS} ${MOD_CL_CPP})
+    # Remove audio augmentations from the list of files to be compiled
     if(NOT RPP_AUDIO_SUPPORT)
         remove_audio_augmentations(Rpp_Source "${BACKEND}")
     endif()

--- a/src/modules/CMakeLists.txt
+++ b/src/modules/CMakeLists.txt
@@ -77,6 +77,31 @@ function(add_kernel_includes KERNEL_FILES)
     configure_file(batch_pd/kernel_includes.cpp.in ${PROJECT_BINARY_DIR}/kernel_includes.cpp)
 endfunction()
 
+function(remove_audio_augmentations SOURCE_LIST BACKEND)
+    if( "${BACKEND}" STREQUAL "HIP")
+        list(REMOVE_ITEM ${SOURCE_LIST}
+            "${CMAKE_CURRENT_SOURCE_DIR}/tensor/hip/kernel/non_silent_region_detection.cpp"
+            "${CMAKE_CURRENT_SOURCE_DIR}/tensor/hip/kernel/to_decibels.cpp"
+            "${CMAKE_CURRENT_SOURCE_DIR}/tensor/hip/kernel/pre_emphasis_filter.cpp"
+            "${CMAKE_CURRENT_SOURCE_DIR}/tensor/hip/kernel/down_mixing.cpp"
+            "${CMAKE_CURRENT_SOURCE_DIR}/tensor/hip/kernel/spectrogram.cpp"
+            "${CMAKE_CURRENT_SOURCE_DIR}/tensor/hip/kernel/resample.cpp"
+            "${CMAKE_CURRENT_SOURCE_DIR}/tensor/hip/kernel/mel_filter_bank.cpp"
+        )
+    endif()
+    list(REMOVE_ITEM ${SOURCE_LIST}
+        "${CMAKE_CURRENT_SOURCE_DIR}/tensor/rppt_tensor_audio_augmentations.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/tensor/cpu/kernel/non_silent_region_detection.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/tensor/cpu/kernel/to_decibels.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/tensor/cpu/kernel/pre_emphasis_filter.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/tensor/cpu/kernel/down_mixing.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/tensor/cpu/kernel/spectrogram.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/tensor/cpu/kernel/resample.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/tensor/cpu/kernel/mel_filter_bank.cpp"
+    )
+    set(${SOURCE_LIST} "${${SOURCE_LIST}}" PARENT_SCOPE)
+endfunction()
+
 function(add_fma_property)
     set_source_files_properties(tensor/rppt_tensor_audio_augmentations.cpp
                                 tensor/rppt_tensor_statistical_operations.cpp
@@ -136,6 +161,10 @@ if( "${BACKEND}" STREQUAL "HIP")
         file(GLOB MOD_CPU_CPP_BATCHPD_KERNELS "legacy_dep/cpu/kernel/*.cpp" )
         list(APPEND Rpp_Source ${MOD_HIP_CPP_BATCHPD_KERNELS} ${MOD_CPU_CPP_BATCHPD_KERNELS})
     endif()
+    # Remove audio augmentations from the list of files to be compiled
+    if(NOT RPP_AUDIO_SUPPORT)
+        remove_audio_augmentations(Rpp_Source "${BACKEND}")
+    endif()
     message("-- ${Green}HIP kernels added${ColourReset}")
 
     # Set compiler flags
@@ -156,6 +185,9 @@ elseif( "${BACKEND}" STREQUAL "OCL")
     file(GLOB MOD_CL_CPP "batch_pd/cl/*.cpp" )
     file(GLOB MOD_HOST_CPP_TENSOR_KERNELS "tensor/cpu/kernel/*.cpp" )
     list(APPEND Rpp_Source ${CPPFILES} ${MOD_HOST_CPP_TENSOR_KERNELS} ${MOD_CL_CPP})
+    if(NOT RPP_AUDIO_SUPPORT)
+        remove_audio_augmentations(Rpp_Source "${BACKEND}")
+    endif()
     message("-- ${Green}OpenCL kernels added!${ColourReset}")
     add_fma_property()
     # Add OpenCL specific preprocessor flags
@@ -175,6 +207,10 @@ elseif( "${BACKEND}" STREQUAL "CPU")
     if(NOT RPP_LEGACY_SUPPORT)
         file(GLOB MOD_CPU_CPP_BATCHPD_KERNELS "legacy_dep/cpu/kernel/*.cpp" )
         list(APPEND Rpp_Source ${MOD_HIP_CPP_BATCHPD_KERNELS} ${MOD_CPU_CPP_BATCHPD_KERNELS})
+    endif()
+    # Remove audio augmentations from the list of files to be compiled
+    if(NOT RPP_AUDIO_SUPPORT)
+        remove_audio_augmentations(Rpp_Source "${BACKEND}")
     endif()
     message("-- ${Green}HOST kernels added${ColourReset}")
     set(INCLUDE_LIST ${CMAKE_SOURCE_DIR}/src/include/common/)

